### PR TITLE
FEAT: Dynamic withdraw amount of Aave yield bearing tokens, Connect Wallet Cta Button, Refactors

### PR DESCRIPTION
**Pull request by t0rbik**
As a user i want to get exactly 1 aUSDC, when I input 1 aUSDC in vault withdraw input, so that i get expected result.

### CODE REVIEW

**Comment by t0rbik** [comment]
I've update this PR with some more changes, so I think we need to do internal testing before we merge into prod.

- Aave yield token withdraw amount should be precisely what we input;
- We change all cta buttons to CtaButton component which triggers and says connect wallet; This is to prevent error when we don't have user connected and they click on cta button.
- More `isPending` and `isFetching` checks let's us remove button flickering. Some context: initially queries sit in `isPending` state (even when `enabled` is set to `false`), so that's why we only take into account config `isPending`, when we're done with approval, otherwise we are always in `isPending` state because configs are disabled before `isApprovalNeeded === false`. For allowance we also check for `isFetching`, because we want to disable button, when we got the approval receipt and triggered query invalidation, which triggers `isFetching=true`. 
Additionally:
before
https://github.com/user-attachments/assets/cdb2d209-8c5b-4519-9653-d9f996d84f3a
after:
https://github.com/user-attachments/assets/dc8de746-dba1-4171-9c6c-68a069a60c73
- Removed redundant `useMemo` and `useCallback`, they didn't give us any perf win, but cost for caching. Moreover, it looks like most of the rest useMemo and useCallback can be as well removed, but I decided to leave it when we do filtering or map arrays. And for functions that could _maybe potentially_ be passed down to some hook dependencies, but again, also subject to removal so we don't waste on memoization.

File: src/hooks/useAllowance.ts
Comment (t0rbik) [inline src/hooks/useAllowance.ts:131]: I'm still hesitant about IIFE in component body but I'm leaning to it, i think it's quite handy when we don't need `useMemo`. But lmk if you think it's not readable. We can change it to
```
let isPending;
if (...) {
isPending = ...
} else etc...
```

**Review by n4n0GH — APPROVED** [review]

**Pull request merged**
